typescript/index.d.ts

```diff
@@ -0,0 +1,264 @@
+// Minimum TypeScript Version: 3.7
+
+// state.d.ts
+export type NodeKey = string;
+export type AtomValues = Map<NodeKey, Loadable<any>>;
+export type ComponentCallback = (state: TreeState) => void;
+export type TreeState = Readonly<{
+  isSnapshot: boolean;
+  transactionMetadata: object;
+  dirtyAtoms: Set<NodeKey>;
+  atomValues: AtomValues;
+  nonvalidatedAtoms: Map<NodeKey, unknown>;
+  nodeDeps: Map<NodeKey, Set<NodeKey>>;
+  nodeToNodeSubscriptions: Map<NodeKey, Set<NodeKey>>;
+  nodeToComponentSubscriptions: Map<NodeKey, Map<number, [string, ComponentCallback]>>;
+}>;
+
+// node.d.ts
+export class DefaultValue {}
+
+// recoilRoot.d.ts
+import * as React from 'react';
+
+export interface RecoilRootProps {
+  initializeState?: (options: {
+      set: <T>(recoilVal: RecoilState<T>, newVal: T) => void;
```

Comment:
nit a few lingering 4 spaces near the top 😃

Reply:
ah good catch! I'll remove